src/main/java/games/strategy/engine/data/GameStep.java

```diff
@@ -18,7 +19,6 @@
   private final String m_displayName;
   private final PlayerID m_player;
   private final String m_delegate;
-  private int m_hashCode = -1;
```

Comment:
I know this class implements Serializeable, but it did work for me removing this param.

Reply:
Apparently, it's acceptable to remove fields from a `Serializable` type and maintain backwards compatibility.  Per the [`ObjectInputStream` spec](https://docs.oracle.com/javase/8/docs/platform/serialization/spec/input.html#a961):

> The `defaultReadObject` method is used to read the fields and object from the stream. ... Any field of the object that does not appear in the stream is set to its default value. **Values that appear in the stream, but not in the object, are discarded.** ...

Reply:
Hmm, actually, I was only thinking in the old -> new direction.  If a new client were to save the game and send it back to an old client, `m_hashCode` wouldn't be present in the stream, and the value would end up defaulting to zero.  That means all `GameStep`s on the old client would have the same hash code upon deserialization.

While that probably doesn't break anything functionally, it will cause all `GameStep`s to be allocated to the same hash bucket, thus making `Map` and `Set` operations fairly inefficient.  Should we just keep this field for the time being?

Reply:
I don't think so. It's going to work, but we should probably note this in the next release notes.
This is just a smaller issue where users just get a performance penalty when going back...